### DISCUSSION TITLE
Sharing about issues related to Booting from SD card and thumb drive

### DISCUSSION THREAD

**Discussion by ductm208** [devmfc/debian-on-amlogic#48] [Q&A]
Hi devmfc and community,

Firstly, I want to use the first sentences to express my appreciation to the team. You are making such efficient and elegant works. Despite some difficulties from making the sdcard from images (which I am sharing now) I can manage to install the Homeassistant on my tvbox to control the IOT devices. Your packages are light weighted but still encapsulate all mini computers need. Comparing 100MB package with 1GB from other armbian ones you are really rock!

So this is my 7-day configuration which is working smoothly:
- Tanix W2 - S905W2: RAM 4GB
- SD card Yoosee 32GB - SDHC U3 - bought last 2 years
- Release: Devmfc_Debian-Bookworm_6.6.23-meson64_Minimal-24.03.29.img.xz

**I also managed to make system boots with:**
- JVJ SDHC Class 6 4GB - bought last year
- Kingston 4GB black thumb drive - bought 10 years back
- No name blue thumb drive 8GB partner gave me last 8 years.

**But it wont boot with:**
- SanDisk SDHC class 10: 32GB - just last week
- Toshiba XDHC U3: 64GB - bought 5 years
- Lexar 32GB thumb drive - just last week

**Here are the accessories: Non functional devices are on the right side.**
![1712633927263](https://github.com/devmfc/debian-on-amlogic/assets/4532530/5d6d149c-c8d5-4705-b2cc-4e5bee42f1e1)

It seems to be kernel issue, it does not support modern storage type. In my observation, this kernel only runs on SDHC class 6 and below. My Yoosee SDHC U3 works but Toshiba XDHC U3 wont regardless they are same size.

This is what I got from HDMI output from non-functional sd cards:
![IMG_4735](https://github.com/devmfc/debian-on-amlogic/assets/4532530/86a193f9-784e-4d34-8e9b-6f91773c99da)

And also from Lexar thumb drive:
![IMG_4737](https://github.com/devmfc/debian-on-amlogic/assets/4532530/11be5acc-cd26-49bb-92f6-0502ad576c9a)

Hope my experiments could help someone, and it would be great if you share what you have faced recently. I am happy with SDHC U3, however better speed should be better experience.

Thanks again for your work devmfc

**Comment by gitisgreat2023** [comment]
Why don't you install to the internal flash drive?
What does free -h show on the 4GB RAM system?
And the first line of 
`cat /proc/meminfo`
MemTotal:

**Reply by ductm208** [reply]
Because if you want to install on emmc you have to make it boot externally first. In my case I am still doing some different experiments, using different sdcards is better than continuously re-install all stuffs in emmc.
Btw this is the output: It seems 3.44GB is provisioned for OS, ~500MB is for GPU.

tvbox:~#cat /proc/meminfo
MemTotal:        3448152 kB
MemFree:           44224 kB
MemAvailable:    1855204 kB
Buffers:          155400 kB
Cached:          1638436 kB
SwapCached:            0 kB
Active:           563964 kB
Inactive:        2581044 kB
Active(anon):      14448 kB
Inactive(anon):  1343316 kB
Active(file):     549516 kB
Inactive(file):  1237728 kB
Unevictable:          80 kB
Mlocked:              80 kB
SwapTotal:        767996 kB
SwapFree:         760060 kB
Zswap:              3760 kB
Zswapped:           7452 kB
Dirty:               340 kB
Writeback:             0 kB
AnonPages:       1322664 kB
Mapped:           452072 kB
Shmem:              7520 kB
KReclaimable:     114104 kB
Slab:             171968 kB
SReclaimable:     114104 kB
SUnreclaim:        57864 kB
KernelStack:       11408 kB
PageTables:        18604 kB
SecPageTables:         0 kB
NFS_Unstable:          0 kB
Bounce:                0 kB
WritebackTmp:          0 kB
CommitLimit:     2492072 kB
Committed_AS:    7534936 kB
VmallocTotal:   133141626880 kB
VmallocUsed:       21588 kB
VmallocChunk:          0 kB
Percpu:             1440 kB
HardwareCorrupted:     0 kB
AnonHugePages:    663552 kB
ShmemHugePages:        0 kB
ShmemPmdMapped:        0 kB
FileHugePages:         0 kB
FilePmdMapped:         0 kB
CmaTotal:         262144 kB
CmaFree:             600 kB
HugePages_Total:       0
HugePages_Free:        0
HugePages_Rsvd:        0
HugePages_Surp:        0
Hugepagesize:       2048 kB
Hugetlb:               0 kB

**Reply by gitisgreat2023** [reply]
My understand was that you can boot from (relatively) slow SD cards and you want to boot from fast SD. You can thus boot once from a slow SD card, install to EMMC and then its fast... or don't I understand your issue?

OK, that's 729MB of reserved memory... that's a lot! Do you have any idea how to get rid of that for a headless server? (no GPU needed there, only ssh access).

**Reply by ductm208** [reply]
I was saying using sdcards is more convenient as I am doing multiple experiments, different setups could be maintained and deployed easily as just plug and play. Installing on emmc I have to erase all stuffs and starting over.

Afaik, to claim back GPU memory reservation dtb need to be compiled again from dtc.

**Reply by gitisgreat2023** [reply]
Yeah, okay, I install and then run the system. With the experimenting its clear why.
"dtb need to be compiled again from dtc" -> what is dtc?

**Reply by devmfc** [reply]
@ductm208  Thanks for your feedback.  

IMHO  sd-cards are a pain in the A...  
I think it is not only hardware, but also the combination of the mainline mmc driver + experimental S4 + wring value in my DTB's...

I always use USB flash disks with Amlogic SoC's, never SD-Cards to experiment.

**Reply by ductm208** [reply]
dtc is the compiler to make dtb, which describes the hardware specs (CPU, ram, clock speeds and peripherals...) to let kernel knows how it could warm up

**Reply by ductm208** [reply]
Yeah I agree Amlogic does not support very well sdcards. I have a thumb-sized board from Luckfox which soldered Rockchip RV1103 on (link as below) but it could boot with any sd cards I have above. 

https://www.luckfox.com/EN-Luckfox-Pico-Plus
